### PR TITLE
Player frame

### PR DESCRIPTION
**Pull request by cjudge1337**
Video details are showing up. Page just needs to be styled now. This should get the basics of this page done. Any additional features will need an OAuth access token to do.

**Before submitting a pull request,** please do the following:

1. Make sure you are not working on the master branch.
2. Push up the branch code and create a pull request.
3. Add details regarding what code you wrote.
4. Select the appropriate label for the pull request
5. If the pull request is related to a particular issue, make sure to reference the issue number.
6. Add [David Hu](https://www.github.com/davidhu2000) as reviewer.

### CODE REVIEW

File: src/components/player/details.jsx [hunk 1]
```diff
@@ -1,11 +1,63 @@
...
+
+    this.state = { details: {} };
+  }
+
+  componentDidMount() {
+    this._fetchDetails();
+  }
+
+  _fetchDetails() {
+    return fetch(`https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${this.props.videoId}&key=${YT_API_KEY.publicDataKey}`)
```
Comment (davidhu2000) [inline src/components/player/details.jsx]: can we throw these fetches into the util file? 

we can just combine `search_result_util.js`, `trending_util.js`, and `player_util.js` into one file. maybe call it `youtube_video_util.js`?

**Review by davidhu2000 — CHANGES_REQUESTED** [review]
let's move the `fetches` into the `util` file. maybe name is `youtube_video_util.js`?

**Comment by cjudge1337** [comment]
Made  a `youtube_video_util.js` file and deleted `player_util.js` since it was pointless. Moved my fetches in there.

**Review by davidhu2000 — APPROVED** [review]

**Comment by davidhu2000** [comment]
is there a reason why you kept the comments, details, and related video in each component's internal state? Instead of creating a `videoDetail` key in the rootReducer and keep the information in the store?

We can minimize the number of API hits if we can store the information in the store, in case the user click back into the same video. If they click a different video, we can check to see if the videoId is different. if so, we fetch the info. Then, we can also do all the fetching in `player/index.jsx` and pass in the info from the store as props to `comments.jsx`, `related.jsx`, and `details.jsx`.

What do you think?

**Note:** I merged the master into this branch to check out the latest changes, make sure you `git pull` before making any changes.

**Review by davidhu2000 — CHANGES_REQUESTED** [review]
is there a reason why you kept the comments, details, and related video in each component's internal state? Instead of creating a `videoDetail` key in the rootReducer and keep the information in the store?

We can minimize the number of API hits if we can store the information in the store, in case the user click back into the same video. If they click a different video, we can check to see if the videoId is different. if so, we fetch the info. Then, we can also do all the fetching in `player/index.jsx` and pass in the info from the store as props to `comments.jsx`, `related.jsx`, and `details.jsx`.

What do you think?

**Note:** I merged the master into this branch to check out the latest changes, make sure you `git pull` before making any changes.

**Comment by cjudge1337** [comment]
I did it mainly because it seemed easier to keep things in the component state instead of the store. Since everything in the video player is independent of the rest of the app, and the only parameter it needs (the video ID) is passed in through the query string, it seemed like overkill at the time to add the store to the mix. Everything will need to be re-fetched for each new video being shown anyway so it seemed like adding extra processing for no real benefit.

However, I didn't think of using it to cut down API hits like you described. It would be a pretty minimal reduction, because the app doesn't have back/forward/refresh buttons like in the browser, so there isn't really a way to make those redundant calls unless they went to the home/search page, then went back to the same video again. Cutting down calls is important though so I can do that if you think it's worth it.

**Comment by davidhu2000** [comment]
we can leave it like this for now. 

Depending on if minimizing API calls is necessary later down the line once we have more features, we can fix this. I will just add an issue so we will remember. 

Merging into master now.

**Review by davidhu2000 — APPROVED** [review]

**Pull request merged**